### PR TITLE
Versions

### PR DESCRIPTION
**Pull request by ivanterekh**
Added version package who's vars are supposed to be assigned during compilation. For this purpose changes were made to Makefile. Version also now apperes as tag in docker image. Version is taken from git as latest tag.

Also made some changes to building docker image flow. Now dependencies are put to vender and then copied to image instead of downloading to image directly.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,19 +2,27 @@ APP ?= bin/go-skeleton
 ENV ?= dev
 PORT ?= 8080
 
+MODULE = github.com/ivanterekh/go-skeleton
+VERSION = $(shell git tag)
```
Comment (obalunenko) [inline Makefile]: suggest to change to 
`$(git describe --tags $(git rev-list --tags --max-count=1))`

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,9 @@
 .idea/
 
+vendor/
```
Comment (obalunenko) [inline .gitignore:3]: https://stackoverflow.com/questions/51118691/should-i-add-go-dep-vendor-to-gitignore

Comment (dzyanis) [reply .gitignore:3]: +1

Comment (ivanterekh) [reply .gitignore:3]: The question you've linked is about go dep. When  using modules it doesn't make any sense:
1. All libraries versions are fixed in `go.mod` and `go.sum`.
2. While using go modules vendor is not automatically updated. So this folder doesn't obviously contain code that is actually used.

As an example hugo framework also has vendor in gitignore.
https://github.com/gohugoio/hugo/blob/master/.gitignore#L13

File: main.go [hunk 3]
```diff
@@ -17,6 +18,12 @@ func main() {
 		log.Fatal(errors.Wrap(err, "could not init logger"))
 	}
 
+	logger.Info("running app",
```
Comment (dzyanis) [inline main.go:21]: As a variant to print this information also on `/health` page. For example https://www.uber.com/health

**Review by dzyanis — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (obalunenko) [reply .gitignore:3]: suppose hugo doesn't use vendor cause they use proxy server for modules
https://docs.gomods.io/

**Pull request merged**